libraries/Microsoft.Bot.Builder.LanguageGeneration/LGFileLexer.g4

```diff
@@ -1,259 +1,24 @@
 lexer grammar LGFileLexer;
 
-// From a multiple-lanague perpective, it's not recommended to use members, predicates and actions to 
-// put target-language-specific code in lexer rules 
-
-// the reason we use it here is that
-// 1. it greatly simplify the lexer rules, can avoid unnecessary lexer modes 
-// 2. it helps us to output token more precisely
-//    (for example, 'CASE:' not followed right after '-' will not be treated as a CASE token)
-// 3. we only use very basic boolen variables, and basic predidates
-//    so it would be very little effort to translate to other languages
-
```

Comment:
Did this all move to the grammar?

Reply:
I see the new LGTemplateLexer/Parser--why do you still need the file one?

---
In reply to: [409983230](https://github.com/microsoft/botbuilder-dotnet/pull/3731#discussion_r409983230) [](ancestors = 409983230)

Reply:
The LGFile lexer\parser are used to correctly parse the file into individual blocks (template, option, comments, import, etc). Without involving too much of the details of how a template is written, it can be more accurate.

The LG template lexer and parser is used to do a thoroughly parsing and examining of the template which extracted from the LG file parser at first step. It can done by safely assume this the body, and give more meaningful messages.